### ISSUE TITLE
[feat req] Delete artifacts when new one is available

### ISSUE THREAD

**Issue by DenizUgur** [actions/upload-artifact#338]
### What would you like to be added?

I would like to remove old artifacts when a new action creates new ones.

### Why is this needed?

Repository size limitations

**Comment by StephenHodgson** [comment]
I agree, it'd be great to configure it to delete the oldest artifact and upload a new one. 
Similar to how the cache system works.

**Comment by DPatrickBoyd** [comment]
doesn't the artifact get overwritten if the name is the same?
